contributors/devel/sig-testing/verify-tests.md

```diff
@@ -105,3 +105,81 @@ Finally the script checks that all external group versions
 (e.g. `foobar/v1`) are defined in `hack/lib/init.sh` in either the
 `KUBE_AVAILABLE_GROUP_VERSIONS` or `KUBE_NONSERVER_GROUP_VERSIONS`
 bash variables.
+
+### `verify-bazel`
+
+This verification script validates whehter the removal of `bazel`
+related files is still needed or not. This script after setting the
+temporary directories for gopath and kube-related path, runs
+`[update-bazel.sh]()`. And once executed, `update-bazel` deletes all
+the bazel related files which are either not there in vendor and some
+additional one-off files. After execution, the `verify-bazel` script
+again calculates diff to check the earlier output. And if the diff
+result is non-zero then script `update-bazel.sh` is run again for the
+files that might be left.  
+
+Following are the files that are searched for deletion as part of this
+script:
+
+```bash
+  "${KUBE_ROOT}/build/root/BUILD.root" \
+  "${KUBE_ROOT}/WORKSPACE" \
+  "${KUBE_ROOT}/build/root/WORKSPACE" \
+  "${KUBE_ROOT}/.bazelrc" \
+  "${KUBE_ROOT}/build/root/.bazelrc" \
+  "${KUBE_ROOT}/.bazelversion" \
+  "${KUBE_ROOT}/build/root/.bazelversion" \
+  "${KUBE_ROOT}/.kazelcfg.json" \
+  "${KUBE_ROOT}/build/root/.kazelcfg.json"
+```
+
+## `verify-boilerplate`
+
+This script checks for the license headers for all the files, whether
+the header is correct or wrong. The purpose of boilerplate headers is
+to identify license headers. The script collects all the file names
+generated by `hack/boilerplate/boilerplate.py` script and stores them
+into a list called `files_need_boilerplate`.
+
+Once we collect all the file names, we run a check to identify the
+files with wrong header. This check will only run if the file exists
+in the list mentioned above.
+
+
+## `verify-cli-conventions`
+
+This script checks whether the description format of `help` message of
+kubectl command is valid or not. This check is done for all the
+`kubectl` sub-commands as well. The script first checks whether the go
+command is available or not in the ${PATH}. And then the binary
+`cmd/clicheck` is checked if it exists or not in the well-known output
+locations. It runs command checks on all the kubectl commands or subcommands like,
+
+```bash
+kubectl version
+kubectl uncordon
+kubectl wait
+kubectl top node
+```
+
+And if the output looks good i.e. CLI follows all tested conventions
+the test passes.
+
+
+## `verify-codegen`
+
+This script verifies if the code update is needed or not for specific
```

Comment:
I think all the "verify-XXX" scripts are merely determining if the "update-XXX" script needs to be run, and this script is no different. Documenting it like this is very confusing. If people want a blow-by-blow description of what it does they should just read the code. High level documentation should explain the purpose, not the steps it takes.

Reply:
@lavalamp thanks for your valuable comments. If possible, can you direct me to some existing example which I can consider while I make some changes to the documentation?